### PR TITLE
🔀 :: (entry-211) 검정고시 산출식 변경

### PR DESCRIPTION
**Pull request by rudeh2926**
image
스크린샷 2024-08-08 오후 11 34 51
스크린샷 2024-08-08 오후 11 33 18

## Summary by CodeRabbit

- **New Features**
	- Introduced a detailed scoring system with individual subject grades (Korean, Social, Math, Science, English, and Optional) for more precise tracking of academic performance.
	- Added functionality to calculate average scores dynamically based on individual subject grades, enhancing accuracy and context relevance.
	- Updated qualifications handling to accommodate multiple grades rather than a single average score.

- **Bug Fixes**
	- Improved logic to ensure accurate retrieval and updating of qualification case data.

### CODE REVIEW

File: application-infrastructure/src/main/kotlin/hs/kr/equus/application/domain/applicationCase/presentation/WebApplicationCaseAdapter.kt [hunk 1]
```diff
@@ -48,7 +48,19 @@ class WebApplicationCaseAdapter(
...
+            request.run {
+                UpdateQualificationCaseRequest(
+                    koreanGrade = koreanGrade,
+                    socialGrade = scienceGrade,
+                    mathGrade = mathGrade,
+                    scienceGrade = scienceGrade,
+                    englishGrade = englishGrade,
+                    optGrade = optGrade,
+                    extraScore = extraScore
+                )
```
[the remark this answers is not in the document]
Comment (rudeh2926) [reply application-infrastructure/src/main/kotlin/hs/kr/equus/application/domain/applicationCase/presentation/WebApplicationCaseAdapter.kt]: 변경했어

File: application-domain/src/main/kotlin/hs/kr/equus/application/domain/applicationCase/model/QualificationCase.kt [hunk 2]
```diff
@@ -10,36 +10,79 @@ data class QualificationCase(
...
-                .setScale(3, RoundingMode.HALF_UP)
-
+        val averageScore = calculatePointAverageScore()
         return if (isCommon) {
-            (totalGradeScore * (COMMON_GRADE_RATE))
-                .setScale(3, RoundingMode.HALF_UP)
+            (averageScore * BigDecimal(34) + BigDecimal(3)).setScale(3, RoundingMode.HALF_UP)
         } else {
-            totalGradeScore
+            (averageScore * BigDecimal(22) + BigDecimal(3) + BigDecimal(6)).setScale(3, RoundingMode.HALF_UP)
```
Comment (HyunSu1768) [inline application-domain/src/main/kotlin/hs/kr/equus/application/domain/applicationCase/model/QualificationCase.kt]: 이 코드에서 3하고 6을 그냥 더하는게 아니라 알고리즘 대회, 자격증 취득을 한 경우에만 더해야 합니다.

File: application-infrastructure/src/main/kotlin/hs/kr/equus/application/domain/applicationCase/presentation/dto/request/UpdateQualificationCaseWebRequest.kt [hunk 3]
```diff
@@ -8,10 +8,11 @@ import javax.validation.constraints.Digits
...
-    @DecimalMax("100.0")
-    @Digits(integer = 3, fraction = 2)
-    @NotNull(message = "ged_average_score는 null이면 안됩니다.")
-    val averageScore: BigDecimal,
+    val koreanGrade: BigDecimal,
+    val socialGrade: BigDecimal,
+    val mathGrade: BigDecimal,
+    val scienceGrade: BigDecimal,
+    val englishGrade: BigDecimal,
+    val optGrade: BigDecimal,
```
Comment (HyunSu1768) [inline application-infrastructure/src/main/kotlin/hs/kr/equus/application/domain/applicationCase/presentation/dto/request/UpdateQualificationCaseWebRequest.kt:16]: 그 전 코드처럼 @DecimalMax("100.0") 를 설정하면 좋을것 같습니다.

File: application-domain/src/main/kotlin/hs/kr/equus/application/domain/applicationCase/model/QualificationCase.kt (hunk 2 above)
Comment (rudeh2926) [reply application-domain/src/main/kotlin/hs/kr/equus/application/domain/applicationCase/model/QualificationCase.kt]: 넵

Comment (rudeh2926) [reply application-domain/src/main/kotlin/hs/kr/equus/application/domain/applicationCase/model/QualificationCase.kt]: [e3948bb](https://github.com/EntryDSM/Equus-Application/pull/64/commits/e3948bb2b2e63805da1755188e8cdbce177c7096)

File: application-infrastructure/src/main/kotlin/hs/kr/equus/application/domain/applicationCase/presentation/dto/request/UpdateQualificationCaseWebRequest.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (rudeh2926) [reply application-infrastructure/src/main/kotlin/hs/kr/equus/application/domain/applicationCase/presentation/dto/request/UpdateQualificationCaseWebRequest.kt]: 변경했어

File: application-domain/src/main/kotlin/hs/kr/equus/application/domain/applicationCase/usecase/dto/request/UpdateQualificationCaseRequest.kt [hunk 4]
```diff
@@ -3,6 +3,11 @@ package hs.kr.equus.application.domain.applicationCase.usecase.dto.request
 import java.math.BigDecimal
 
 data class UpdateQualificationCaseRequest(
-    val averageScore: BigDecimal = BigDecimal(0),
+    val koreanGrade: BigDecimal,
+    val socialGrade: BigDecimal,
+    val mathGrade: BigDecimal,
+    val scienceGrade: BigDecimal,
+    val englishGrade: BigDecimal,
+    val optGrade: BigDecimal,
```
[the remark this answers is not in the document]
Comment (rudeh2926) [reply application-domain/src/main/kotlin/hs/kr/equus/application/domain/applicationCase/usecase/dto/request/UpdateQualificationCaseRequest.kt]: 변경했어

**Review by HyunSu1768 — APPROVED** [review]

**Pull request merged**